plugins/modules/ndo_l3out_node_group_policy.py

```diff
@@ -0,0 +1,333 @@
+    choices: [ absent, query, present ]
+    default: query
+notes:
+- The O(template) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_template) to create the L3Out template.
+- The O(l3out) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_l3out_template) to create the L3Out object under the L3Out template.
+- The O(node_routing_policy) must exist before using this module in your playbook.
+  Use M(cisco.mso.ndo_l3out_node_routing_policy) to create the L3Out Node Routing Policy.
+extends_documentation_fragment: cisco.mso.modules
+"""
+
+EXAMPLES = r"""
+- name: Create a new L3Out node group policy
+  cisco.mso.ndo_l3out_node_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: node_group_policy_1
+    state: present
+
+- name: Update an existing L3Out node group policy
+  cisco.mso.ndo_l3out_node_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: node_group_policy_1
+    description: "Updated description"
+    node_routing_policy: ans_node_policy_group_1
+    bfd_multi_hop_authentication: enabled
+    bfd_multi_hop_key_id: 1
+    bfd_multi_hop_key: TestKey
+    target_dscp: af11
+    state: present
+
+- name: Query a L3Out node group policy
+  cisco.mso.ndo_l3out_node_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: node_group_policy_1
+    state: query
+  register: query_with_name
+
+- name: Delete an existing L3Out node group policy with name
+  cisco.mso.ndo_l3out_node_group_policy:
+    host: mso_host
+    username: admin
+    password: SomeSecretPassword
+    template: l3out_template
+    l3out: l3out
+    name: node_group_policy_1
+    state: absent
+"""
+
+RETURN = r"""
+"""
+
+
+import copy
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.mso.plugins.module_utils.mso import MSOModule, mso_argument_spec
+from ansible_collections.cisco.mso.plugins.module_utils.template import MSOTemplate, KVPair
+from ansible_collections.cisco.mso.plugins.module_utils.constants import TARGET_DSCP_MAP
+from ansible_collections.cisco.mso.plugins.module_utils.utils import generate_api_endpoint
+
+
+def main():
+    argument_spec = mso_argument_spec()
+    argument_spec.update(
+        template=dict(type="str", required=True, aliases=["l3out_template"]),
+        l3out=dict(type="str", required=True, aliases=["l3out_name"]),
+        name=dict(type="str", aliases=["l3out_node_group_policy"]),
+        description=dict(type="str"),
+        node_routing_policy=dict(type="str"),
+        bfd_multi_hop_authentication=dict(type="str", choices=["enabled", "disabled"]),
+        bfd_multi_hop_key_id=dict(type="int"),
+        bfd_multi_hop_key=dict(type="str", no_log=False),
```

Comment:
When we set to no_log, should we also remove from the current/proposed output?

Reply:
The `no_log` is used to avoid sanity failure, but `bfd_multi_hop_key` is not a sensitive attribute.